plugin.video.themoviedb.helper/resources/lib/requestapi.py

```diff
@@ -0,0 +1,146 @@
+        _xbmcdialog.ok('Missing/Invalid {0} API Key'.format(api_name),
+                       'You must enter a valid {0} API key to use this add-on'.format(api_name))
+        xbmc.executebuiltin('Addon.OpenSettings({0})'.format(self.addon_name))
+
+    def get_cache(self, cache_name):
+        return _cache.get(cache_name)
+
+    def set_cache(self, my_object, cache_name, cache_days=14):
+        if my_object and cache_name and cache_days:
+            _cache.set(cache_name, my_object, expiration=datetime.timedelta(days=cache_days))
+        return my_object
+
+    def use_cache(self, func, *args, **kwargs):
+        """
+        Simplecache takes func with args and kwargs
+        Returns the cached item if it exists otherwise does the function
+        """
+        cache_days = kwargs.pop('cache_days', 14)
+        cache_name = kwargs.pop('cache_name', self.addon_name)
+        cache_only = kwargs.pop('cache_only', False)
+        for arg in args:
+            if arg:
+                cache_name = u'{0}/{1}'.format(cache_name, arg)
+        for key, value in kwargs.items():
+            if value:
+                cache_name = u'{0}&{1}={2}'.format(cache_name, key, value)
+        my_cache = self.get_cache(cache_name)
+        if my_cache:
+            return my_cache
+        elif not cache_only:
+            my_object = func(*args, **kwargs)
+            return self.set_cache(my_object, cache_name, cache_days)
+
+    def rate_limiter(self, wait_time=None, api_name=None):
+        """
+        Simple rate limiter to prevent overloading APIs
+        """
+        wait_time = wait_time if wait_time else 2
+        api_name = api_name if api_name else ''
+        nart_time_id = '{0}{1}.nart_time_id'.format(self.addon_name, api_name)
+        nart_lock_id = '{0}{1}.nart_lock_id'.format(self.addon_name, api_name)
+        nart_time = xbmcgui.Window(10000).getProperty(nart_time_id)  # Get our saved time value
+        nart_time = float(nart_time) if nart_time else -1  # If no value set to -1 to skip rate limiter
+        nart_time = nart_time - time.time()
+        if nart_time > 0:  # Apply rate limiting if next allowed request time is still in the furture
+            nart_lock = xbmcgui.Window(10000).getProperty(nart_lock_id)
+            while nart_lock == 'True':  # If another instance is applying rate limiting then wait till it finishes
+                time.sleep(1)
+                nart_lock = xbmcgui.Window(10000).getProperty(nart_lock_id)
+            nart_time = xbmcgui.Window(10000).getProperty(nart_time_id)  # Get nart again because it might have elapsed
+            nart_time = float(nart_time) if nart_time else -1
+            nart_time = nart_time - time.time()
+            if nart_time > 0:  # If nart still in the future then apply rate limiting
+                xbmcgui.Window(10000).setProperty(nart_lock_id, 'True')  # Set the lock to prevent concurrent limiters
+                while nart_time > 0:
```

Comment:
You should consider using OnAbortRequested() here, however, I'm okay with using `if xbmc.Monitor().waitForAbort(1): break`
at line 79.
https://codedocs.xyz/xbmc/xbmc/class_x_b_m_c_addon_1_1xbmc_1_1_monitor.html

Reply:
To rectify this issue, I used `while not xbmc.Monitor().abortRequested():` here instead as docs state that `OnAbortRequested()` is depreciated.

Reply:
My bad got the two mixed :)